compiler/rustc_middle/src/ty/impls_ty.rs

```diff
@@ -135,8 +135,8 @@ impl<'a> HashStable<StableHashingContext<'a>> for ty::RegionKind {
             ty::RePlaceholder(p) => {
                 p.hash_stable(hcx, hasher);
             }
-            ty::ReVar(..) => {
-                bug!("StableHasher: unexpected region {:?}", *self)
+            ty::ReVar(reg) => {
+                reg.hash_stable(hcx, hasher);
```

Comment:
I wonder if the LLVM failure  is due to this. As a refresher, what are the failures when you keep the `bug!`?

Reply:
Don't think it's related to that. `ReVar` is `HashStable`, we just previously had the `bug!` call in there, because it wasn't used when hashing `Region` afaict. I just got the output of the message in the bug statement.

I'm sort of leaning towards thinking that this failure is spurious. The test is in the ui tests and passed on all other platforms and the code changes in this PR don't touch anything that should cause different behaviour on different platforms.